beacon_node/lighthouse_network/gossipsub/src/behaviour.rs

```diff
@@ -2751,6 +2767,16 @@ where
                     .entry(*peer_id)
                     .or_default()
                     .non_priority += 1;
+            } else {
```

Comment:
nit: can we return early instead of using `else`?  So that the below part of code is on the main scope of the function and it's clearer that it is a success case